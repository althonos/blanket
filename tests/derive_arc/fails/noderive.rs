extern crate impls;
extern crate static_assertions;

use std::sync::Arc;
use std::sync::atomic::AtomicU8;
use std::sync::atomic::Ordering;

use impls::impls;
use static_assertions::const_assert;

pub trait Counter {
    fn increment(&self);
}

#[derive(Default)]
struct AtomicCounter {
    count: AtomicU8
}

impl Counter for AtomicCounter {
    fn increment(&self) {
        self.count.fetch_add(1, Ordering::SeqCst);
    }
}

fn main() {
    const_assert!(impls!(AtomicCounter:       Counter));
    const_assert!(impls!(Arc<AtomicCounter>:  Counter));
}
