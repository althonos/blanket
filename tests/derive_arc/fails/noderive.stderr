error[E0080]: evaluation of constant value failed
  --> tests/derive_arc/fails/noderive.rs:28:5
   |
28 |     const_assert!(impls!(Arc<AtomicCounter>:  Counter));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ attempt to compute `0_usize - 1_usize`, which would overflow
   |
   = note: this error originates in the macro `const_assert` (in Nightly builds, run with -Z macro-backtrace for more info)
